base/reflection.jl

```diff
@@ -843,19 +843,21 @@ function MethodList(mt::Core.MethodTable)
 end
 
 """
-    methods(f, [types])
+    methods(f, [types], [module])
 
-Returns the method table for `f`.
+Return the method table for `f`.
 
-If `types` is specified, returns an array of methods whose types match.
+If `types` is specified, return an array of methods whose types match.
+If `module` is specified, return an array of methods defined in this module.
 """
-function methods(@nospecialize(f), @nospecialize(t))
+function methods(@nospecialize(f), @nospecialize(t), mod::Union{Module,Nothing}=nothing)
     if isa(f, Core.Builtin)
         throw(ArgumentError("argument is not a generic function"))
     end
     t = to_tuple_type(t)
     world = typemax(UInt)
-    return MethodList(Method[m[3] for m in _methods(f, t, -1, world)], typeof(f).name.mt)
+    MethodList(Method[m[3] for m in _methods(f, t, -1, world) if mod === nothing || m[3].module == mod],
```

Comment:
Could arguably be ` parentmodule(m[3].module)`.

Reply:
Why? If I want methods defined in a submodule `S` of `P`, then `methods(f, S)` will give nothing and `methods(f, P)` will give all methods of `f` whether they are in `S` or `P`...

Reply:
Yeah, I think I meant `m[3].module == mod || parentmodule(m[3].module) == mod`.

Basically, if you are asking for the methods in `Pkg` you probably want the method in `Pkg.API` as well? Anyway, doesn't have to be done now.

Reply:
Yes a "recursive" option would be useful, maybe as a keyword.